File: articles/numerical-methods-and-algorithmic-approximation/cpp/numerical_approximation.cpp
#include <cmath>
#include <iomanip>
#include <iostream>

double f(double x) { return std::sin(x) + 0.25 * x * x; }
double central_difference(double x, double h) { return (f(x + h) - f(x - h)) / (2.0 * h); }
int main() { std::cout << std::setprecision(12) << central_difference(1.0, 0.01) << "\n"; }

File: articles/numerical-methods-and-algorithmic-approximation/rust/numerical_approximation.rs
fn f(x: f64) -> f64 { x.sin() + 0.25 * x * x }
fn central_difference(x: f64, h: f64) -> f64 { (f(x + h) - f(x - h)) / (2.0 * h) }
fn main() { println!("{:.12}", central_difference(1.0, 0.01)); }
